File: level_1/matrix_layerwise_rotation/langs/compiler.py
def main():
    m = int(input())
    n = int(input())

    input_mat = [[0 for i in range(n)] for i in range(m)]
    invalid = False

    for i in range(m):
        num_elements_in_a_row = 0
        row = input()
        tokens = row.split(' ')
        for j, val in enumerate(tokens):
            input_mat[i][j] = int(val)
            num_elements_in_a_row += 1

        # If any row has lesser/greater number of elements than n, the input is invalid
        if num_elements_in_a_row != n:
            invalid = True
            return

    # Check if the input is valid
    if invalid:
        print("You have entered an invalid input!")
        return

    output_mat = layerwise_rotate(input_mat)
    # Print the sorted matrix
    for i in range(n):
        for j in range(m):
            print(output_mat[i][j], end=' ')
        print()

if __name__ == '__main__':
    main()

File: level_1/matrix_layerwise_rotation/langs/compiler.c
int main()
{
    int m;
    int n;
    int **input;
    scanf("%d", &m);
    scanf("%d", &n);

    input = (int **)malloc(m * sizeof(int*));
    for (int i = 0; i < m; i++)
        input[i] = (int *)malloc(n * sizeof(int));

    char temp;
    char row[MAX_CHARS_PER_LINE];
    row[0] = '\0';
    int invalid = 0;
    for (int i = 0; i < m; i++)
    {
        int num_elements_in_a_row = 0;
        scanf("%c", &temp);
        scanf("%[^\n]", row);
        // Check if the string/row is empty
        if (row[0] == '\0')
        {
            invalid = 1;
            break;
        }

        char *tok;
        tok = strtok(row, " ");
        while (tok != NULL)
        {
            input[i][num_elements_in_a_row] = atoi(tok);
            num_elements_in_a_row++;
            tok = strtok(NULL, " ");
        }
        // If any row has lesser/greater number of elements than n, the input is invalid
        if (num_elements_in_a_row != n)
            invalid = 1;
    }

    // Check if the input is valid
    if (invalid)
    {
        printf("You have entered an invalid input!\n");
        return 1;
    }

    int **output = (int **)calloc(n, sizeof(int*));
    for (int i = 0; i < n; i++)
        output[i] = (int *)calloc(m, sizeof(int));

    layerwise_rotate(input, output, m, n);

    // Print the sorted matrix
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            printf("%d ", output[i][j]);
        }
        printf("\n");
    }

    // Free memory
    for (int i = 0; i < m; i++)
        free(input[i]);
    for (int i = 0; i < n; i++)
        free(output[i]);
    free(input);
    free(output);

    return 0;
}